### ISSUE TITLE
[Feature Request] Band Effect: Filter (pre-drive)

### ISSUE THREAD

**Issue by Phazmaonline** [jerryuhoo/Fire#36]
I think it could be a cool idea to have a Filter also as Band Effect, with the possibility to filter before distortion. Having a resonant filter going into the distortion and automating the cutoff frequency can give a very nice sweeping effect, specially on basslines. (A tempo-synced LFO to control it would be the cherry on top but maybe that would go a bit too far for what you intended Fire to be.)

**Comment by jerryuhoo** [comment]
Do you mean a global filter before distortion or a filter for each band?
```
               -> band 1 distortion ->
a). pre-filter -> band 2 distortion -> post-filter
               -> band 3 distortion ->

    pre-filter ->  band 1 distortion ->
b). pre-filter ->  band 2 distortion -> post-filter
    pre-filter ->  band 3 distortion ->
```

**Comment by jerryuhoo** [comment]
> (A tempo-synced LFO to control it would be the cherry on top but maybe that would go a bit too far for what you intended Fire to be.) 

Designing a tempo-synced LFO to control is very complicated, and I don't think I can make that with my current skills. 😂

**Comment by Phazmaonline** [comment]
Originally, for flexibility, I thought of your suggestion "b" (filter for each band). 
But now that you mention it, actually I think suggestion "a" (global pre-dist filter) is probably easier to control and get useful results with. It would certainly be sufficient for my kind of use.

**Comment by jerryuhoo** [comment]
Yeah, I also think "a" is easier to design, I would use "a".

**Comment by Phazmaonline** [comment]
Ok, if you manage to add option "a" that would be very cool :)

**Comment by jerryuhoo** [comment]
I'm kinda busy these days so this update will not be available for v1.0.0 and will appear in the next version.

**Comment by Phazmaonline** [comment]
No worries, thanks meanwhile for the other implementations! Btw I tested v1.0.0 and commented my thoughts on the other 2 closed issues (#35 and #37 ).

**Comment by IsaakCode** [comment]
Yes, one global pre-saturation stage filter would be perfect (similar to the current implementation of the post-saturation EQ, just pre).

**Comment by jerryuhoo** [comment]
I'm revisiting this issue. The current version (actually should be one of the previous releases) already supports a global filter! But the current version also added a value to show the hz and dB!

**Comment by jerryuhoo** [comment]
Oh my mistakes, it's a PRE filter. I will see if I have time to implement it (with AI now it should be fast compared to 3 years ago😊).
